### PR TITLE
feat(on-ramp): upgrade on-ramp-sdk to v1.22.0 with abort controller support

### PR DESCRIPTION
**Pull request by wachunei**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

This PR upgrades `on-ramp-sdk` to v1.22.0 and adopts `AbortController` support for the `useSDKMethod` hook

**Screenshots/Recordings**

~~_If applicable, add screenshots and/or recordings to visualize the before and after of your change_~~

**Issue**

Progresses #???

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

**Comment by wachunei** [comment]
@SocketSecurity ignore-all

File: app/components/UI/FiatOnRampAggregator/hooks/useSDKMethod.ts [hunk 1]
```diff
@@ -30,6 +30,24 @@ function validMethodParams<T extends keyof RegionsService>(
   });
 }
 
+/**
+ * Determines if the provided method is called with all its parameters.
+ * @param method - The method to check the parameters against.
+ * @param params - The parameters to check against the RegionsService interface.
+ * @returns A boolean indicating whether or not the provided parameters are all the parameters of the specified method.
+ */
+function hasAllParams<T extends keyof RegionsService>(
```
Comment (sethkfman) [inline app/components/UI/FiatOnRampAggregator/hooks/useSDKMethod.ts:39]: Can you add a test case?

Comment (sethkfman) [reply app/components/UI/FiatOnRampAggregator/hooks/useSDKMethod.ts:39]: Planned in a separate PR initiative for hooks

**Review by sethkfman — APPROVED** [review]
LGTM

**Comment by bkirb** [comment]
@wachunei LGTM I finished verifying this change

**Pull request merged**